src/test/java/com/uid2/admin/util/PrivateSiteUtilTest.java

```diff
@@ -589,6 +590,77 @@ public void testGenerateEncryptionKeyAclDataForEachSite() {
 
     }
 
+    @Nested
+    class ClientSideKeypairs {
+
+        ClientSideKeypair pair1 = new ClientSideKeypair("0123456789", new byte[] {0, 1,2 }, new byte[] {0, 1, 2}, 123, "contact@email.com", Instant.now(), false, "UID2-X-T-", "UID2-Y-T-");
+        ClientSideKeypair pair2 = new ClientSideKeypair("0123456789", new byte[] {0, 1,2 }, new byte[] {0, 1, 2}, 123, "contact@email.com", Instant.now(), false, "UID2-X-T-", "UID2-Y-T-");
+        ClientSideKeypair pair3 = new ClientSideKeypair("0123456789", new byte[] {0, 1,2 }, new byte[] {0, 1, 2}, 124, "contact@email.com", Instant.now(), false, "UID2-X-T-", "UID2-Y-T-");
+
+        Collection<ClientSideKeypair> keypairs = new ArrayList<>() {{
+           add(pair1);
+           add(pair2);
+           add(pair3);
+        }};
+
+        Map<Integer, Keyset> keysets = new HashMap<Integer, Keyset>();
+
+        @Test
+        public void returnsNoKeysetKeysForNoKeysetKeysOrOperators() {
+            PrivateSiteDataMap<ClientSideKeypair> expected = new PrivateSiteDataMap<>();
+
+            PrivateSiteDataMap<ClientSideKeypair> actual = getClientSideKeypairs(noOperators, noKeypairs);
+
+            assertEquals(expected, actual);
+        }
+
+        @Test
+        public void ignoresSitesWithOnlyPublicOperators() {
+
+            PrivateSiteDataMap<ClientSideKeypair> expected = new PrivateSiteDataMap<>();
+
+            ImmutableList<OperatorKey> operators = ImmutableList.of(
+                    new OperatorBuilder()
+                            .withSiteId(siteId1)
+                            .withType(OperatorType.PUBLIC)
+                            .build()
+            );
+
+            PrivateSiteDataMap<ClientSideKeypair> actual = getClientSideKeypairs(
+                    operators,
+                    keypairs
+            );
+
+            assertEquals(expected, actual);
+        }
+
+        @Test
+        public void siteHasOwnKeypairs() {
+
+            PrivateSiteDataMap<ClientSideKeypair> expected = new PrivateSiteDataMap<>();
+            expected.put(123, ImmutableSet.of(pair1, pair2));
+            expected.put(124, ImmutableSet.of(pair3));
+
+            ImmutableList<OperatorKey> operators = ImmutableList.of(
+                    new OperatorBuilder()
+                            .withSiteId(123)
+                            .withType(OperatorType.PRIVATE)
+                            .build(),
+                    new OperatorBuilder()
+                            .withSiteId(124)
+                            .withType(OperatorType.PRIVATE)
+                            .build()
+            );
+
+            PrivateSiteDataMap<ClientSideKeypair> actual = getClientSideKeypairs(
```

Comment:
you may keep these codes but currently we are not going to enable CSTG for private operators for now. So we do not need to provide any data in getClientSideKeypairs for private oeprator for now. 

We either simply don't generate site-specific data for keypair data (always keep keypair data as GlobalScope not SiteScope), or always return empty data for getClientSideKeypairs method.

Reply:
Ok I disabled this with a comment in PrivateSiteDataSyncJob.java
`//        keypairSyncJob.execute(); Keypair sync Job should not be executed until we want to enable CSTG for private operators`
if that works for you.

Reply:
looks good!